runtime/compiler/env/J9CompilerEnv.cpp

```diff
@@ -66,3 +69,39 @@ J9::CompilerEnv::isCodeTossed()
 
    return false;
    }
+
+TR::PersistentAllocator &
+J9::CompilerEnv::persistentAllocator()
+   {
+#if defined(J9VM_OPT_JITSERVER)
```

Comment:
As I mentioned in https://github.com/eclipse/omr/pull/5760 we should add a simple test to detect JITServer and eliminate TLS access overhead for non-jitserver.
The simplest approach is to make `PersistentInfo::_remoteCompilationMode` static because we already use it, it is set very early and never changes after it is set.

Reply:
Yes, I'll do it.
This made me realize that I need to deal with `compInfo->persistentMemory()` as well, since currently it always returns the global memory.
JITServer only retrieves persistent memory from compInfo in a couple of places, so I'm thinking to just replace them with calls to `TR::Compiler->persistentMemory()` and leave the `compInfo` memory alone, as it's also used to get persistent info and we want that to remain global.